FWCore/Framework/interface/produce_helpers.h

```diff
@@ -46,12 +46,18 @@ namespace edm {
          template< typename T> struct product_traits<T*> {
             typedef EndList<T*> type;
          };
+         template< typename T> struct product_traits<std::unique_ptr<T> > {
```

Comment:
I certainly do want this, however I think just declaring this will be insufficient. We'd need a test to see if it will work.